go-controller/pkg/metrics/master.go

```diff
@@ -485,16 +489,32 @@ const (
 	// port binding with updated chassis seen in OVN-Kubernetes  control plane southbound database libovsdb cache
 	portBindingChassis
 	portBindingTable = "Port_Binding"
+
+	queueCheckPeriod           = time.Millisecond * 500
+	addPortBinding   operation = iota
+	updatePortBinding
+	addPod
+	cleanPod
+	addLogicalSwitchPort
 )
 
 type record struct {
 	timestamp time.Time
 	timestampType
 }
 
+// item is the data structure for ControlPlaneRecorders queue
+type item struct {
+	op     operation
+	old    *model.Model
+	new    *model.Model
+	podUID kapimtypes.UID
+}
+
 type ControlPlaneRecorder struct {
-	sync.Mutex
 	podRecords map[kapimtypes.UID]*record
+	queue      workqueue.Interface
+	queueReady bool
```

Comment:
I don't think we need this, the workqueue should already handle what happens when adding items on a shutdown queue. Or am I missing something?

Reply:
You're right on shutdown. Internally within the queue, it will handle things fine. 
But because we do not create the queue until Run is called (so that non-leaders dont have useless allocation of a queue), if anyone called the public Methods before Run is called, then it will cause a segfault.
This will happen on numerous of our test cases.

Reply:
Ah ok. No strong opinion but can't we just check for nil on the queue then?

Reply:
Yeah. That's better. I'll change it.

Reply:
Another thing, since that variable (`queue` or `queueReady` for this matter) is accessed from multiple goroutines and is not synchronized, it may not be observed to change at the moment it really does from other goroutines. I don't think that is really important here, but you might have another opinion.